tests/unit/test_auth.py

```diff
@@ -59,3 +91,93 @@ def test_generate_consumer_acls():
 
     assert sorted(operations) == sorted(set(["READ", "DESCRIBE"]))
     assert sorted(resource_types) == sorted(set(["TOPIC", "GROUP"]))
+
+
+def test_get_acls_tls_adds_zk_tls_flag(harness):
+    peer_rel_id = harness.add_relation(PEER, CHARM_KEY)
+    zk_rel_id = harness.add_relation(ZK, ZK)
+    harness.add_relation_unit(zk_rel_id, "zookeeper/0")
+    harness.update_relation_data(
+        zk_rel_id,
+        ZK,
+        {
+            "username": "relation-1",
+            "password": "mellon",
+            "endpoints": "123.123.123",
+            "chroot": "/kafka",
+            "uris": "123.123.123/kafka",
+            "tls": "enabled",
+        },
+    )
+    harness.update_relation_data(peer_rel_id, CHARM_KEY, {"tls": "enabled"})
+    auth = KafkaAuth(harness.charm, opts=["mordor"], zookeeper="server.1:gandalf.the.grey")
```

Comment:
love this 🧝‍♀️

Reply:
I was especially proud of https://github.com/canonical/kafka-operator/blob/672a97982fe2b439fad5d84c331236ea7a58f117/tests/unit/test_charm.py#L277-L284

He went from grey to white! config has changed.